### DISCUSSION TITLE
v1.25 feedback thread

### DISCUSSION THREAD

**Discussion by Lailloken** [Lailloken/Exile-UI#62] [Ideas & Feedback]
Highlighted changes:
- added a recombinator calculator
- added a search-strings feature
- reworked & improved the betrayal info-sheet
- added windowed-mode support

**Comment by OmikronGamma** [comment]
Hi again,
I dont know how fixable this is, but there is a  flaw with recomb calc. I have these two bows, [One](https://i.imgur.com/vxl6a5B.png) and [Two](https://i.imgur.com/5WDWh5P.png). 
[This](https://i.imgur.com/x4SgjxG.png) is the result of the calc. As we can see, it cant distinguish between %PhysDmg and %PhysDMg+#Acc mods; and it shows 100% chance for success, which is optimistic, to say the least.

[This other calc](https://www.reddit.com/r/pathofexile/comments/v4yqg5/recombinator_calculator/) shows 20% for success, which is closer actual odds, i believe.

**Reply by Lailloken** [reply]
I must have overlooked something because it can distinguish them on the items themselves (the edit fields have the mods loaded correctly) but not in the pool for desired mods. I'll look into it shortly.

**Reply by Lailloken** [reply]
The bug itself is fixed in v1.25.1-hotfix, but my calculations are a bit more optimistic in certain cases than the google doc. I'll look into that tomorrow. The author seems to have changed something in the formulas without documenting it, and I can't tell off the top of my head if it's 100% accurate now. Last time I checked, the doc couldn't replicate all the examples from the guide, so I made some deductions on my own which may not apply to all cases.

**Comment by OmikronGamma** [comment]
Thanks for looking into it.

> my calculations are a bit more optimistic in certain cases than the google doc.

I was hesitant to bring this up, because i wasnt sure if you are willing to look into it, because who knows how the author made that calc (i guess its google doc, formulas are in there, but still).

**Reply by Lailloken** [reply]
I have just released v1.25.2 with fixes and improvements to the recombinator calculator. Feel free to check it out.

**Comment by OmikronGamma** [comment]
Bug report.
Omni-menu on Ward items leads to a wrong pages - to a general boots/helmet/gloves sections on both PoE Wiki and PoEDB.
It should lead to [Here for Gloves](https://poedb.tw/us/Runic_Gauntlets#ModifiersCalc), [Here for Helmet](https://poedb.tw/us/Runic_Crown#ModifiersCalc) and [Here for Boots](https://poedb.tw/us/Runic_Sabatons#ModifiersCalc) on PoEDB.
As for PoeWiki, [Here](https://www.poewiki.net/wiki/Runic_base_type) is a page with all the ward bases, and i guess the tool should lead to each base's page? [Runic Crown](https://www.poewiki.net/wiki/Runic_Crown), [Runic Crest](https://www.poewiki.net/wiki/Runic_Crest), [Runic Helm](https://www.poewiki.net/wiki/Runic_Helm)

**Reply by Lailloken** [reply]
Whoops. The code is from before expedition league, and I never added support for ward items because that was a league I didn't get far and didn't get to use them (I was playing HC back then, and I think it was the big nerf patch). I meant to add support once I knew they're going core but then apparently forgot (because I also haven't used them since).

Thanks for all your help, appreciate it. I'll add support in the next release.

**Reply by Lailloken** [reply]
v1.25.4 now has support for ward item bases. One thing is different, though: The wiki shortcut will show the page with all bases. This is because the idea behind the 'wiki (item class)' shortcut is to check if a given item is the best in class. I sometimes need that at the beginning of the league when finding potential crafting bases and not having set up a custom item filter. So when finding a somewhat high-lvl base, I check where it ranks in comparison to others within the class to evaluate if I can start crafting on it or if I should wait for a better base.

The page with ward bases is also not too long, so that should not really be a problem.

**Comment by g0sy** [comment]
Hello there! I'm glad I found this, very helpfull tool! I have one problem tho. The Screen resolution. I'm playing on a ultra wide screen and the resolution options on the gui doesn't let me apply that. I have a custom resolutio with 3400x1160, but there are no options to change the first value of the resolution, just the 2nd, and even them are fixed values. Would It be possible for a "free" resolution change in the future updates?

**Reply by OmikronGamma** [reply]
https://www.reddit.com/r/pathofexile/comments/virkct/refining_the_concept_of_cheatsheets_no_alttabbing/idesd90/

Maybe this will shed some light on your question.

**Reply by Lailloken** [reply]
I am currently working on support for windowed mode and custom resolutions, and one thing is variable horizontal resolution. The reason the vertical resolution is fixed is that the game-client behaves inconsistently with non-standard or uncommon vertical resolutions. At these resolutions, some UI elements render incorrectly, which heavily influences any feature that uses image-recognition.

For example, I had an archnemesis feature that scanned the inventory and showed you recipes that were available. And at resolutions such as 768p, 1024p, and 1050p, the same icon would render differently depending on in which slot it was. I tried to fix that by making image-recognition less strict, but some icons were just 're-skinned' or re-colored versions of another icon, which led to incorrect scans and recipe results. And that's actually why the custom resolution feature is in my tool in the first place. It's there to offer those users an alternative so they can still use the tool.

That's why the tool is so strict about resolution in its current iteration. The available vertical resolutions are resolutions that I have tested and don't exhibit this render inconsistency in the game-client. It would probably be fine to change that right now, but if next league's mechanic had opportunities for image-recognition, I would need to put restrictions on those users again and take something away from them (whereas now I'm offering them an alternative to still be able to use the tool).


> I'm playing on a ultra wide screen and the resolution options on the gui doesn't let me apply that. I have a custom resolutio with 3400x1160

What is your monitor's native resolution and how did you usually apply this custom resolution? Just by resizing the window, or did you use another tool? I need some information that I can use to integrate windowed mode support. What I have working right now is running a normal windowed PoE-client at a supported vertical resolution. I also managed to make the tool apply a custom horizontal and supported custom vertical resolution to a maximized window.

In the next release, you would need to either run 3400x1200 or 3400x900 since those two are the closest supported vertical resolutions.

**Reply by g0sy** [reply]
I see, this is a great tool and you have done a great job with this! I truely hope you can find a solution for the manual resolution resizing.
I manually resized the window to my liking. And it stays the same, I dont fiddle with it at all.

**Reply by Lailloken** [reply]
v1.25.5 now supports windowed-mode, feel free to try it out. As mentioned before, you will have to run the client at 3400x1200 as that's the closest officially-supported vertical resolution.
